ops/framework.py

```diff
@@ -330,6 +330,21 @@ class Object(metaclass=_Metaclass):
     handle_kind = HandleKind()
 
     def __init__(self, parent, key):
+        """Initialize the Object, tracking the framework reference and the object's unique name.
```

Comment:
I wonder about the wording and how to give a clear understanding of the design.

OF supports the premise of multiple Frameworks being instantiated (which is very useful in test suites, etc).
Every Object belongs to *one* framework
Every Object has a parent (which might be the framework), which then defines a "path to the object")
eg
```
>>> import ops.storage, ops.framework
>>> store = ops.storage.SQLiteStorage("test.db")
>>> framework = ops.framework.Framework(store, None, None, None)
>>> class Test(os.framework.Object):
...   handle_kind = 'test'
...
>>> a = Test(framework, 'a')
>>> b = Test(a, 'b')
>>> a.handle.path
'test[a]'
>>> b.handle.path
'test[a]/test[b]
```

The other bit is that we both track a reference to the Framework, but the Framework also starts tracking this object. (it is one way that we can sent events to them, and also to ensure that you don't have 2 objects with the same path)

Reply:
Maybe:

```
Initialize an Object as a new leaf in :class:`Framework`, identified by `key`
```

This presumes that the reader is familiar with k/v store leaf tracking, but trees are a common enough datastructure that using that terminology may make it more identifiable